test/gtest/ucp/test_ucp_mmap.cc

```diff
@@ -547,7 +552,11 @@ UCS_TEST_P(test_ucp_mmap, reg_mem_type) {
 
     for (int i = 0; i < 1000 / ucs::test_time_multiplier(); ++i) {
         size_t size    = ucs::rand() % UCS_MBYTE;
-        alloc_mem_type = mem_types.at(ucs::rand() % mem_types.size());
+        auto flags     = mem_map_flags();
+        /* Test ODP registration with host buffer only */
+        alloc_mem_type = (flags & VARIANT_MAP_NONBLOCK) ?
+                UCS_MEMORY_TYPE_HOST :
```

Comment:
so the issue with cuda mem still exists right?

Reply:
1. we should randomize the memory type based on md_attr.reg_nb_mem_types, not just assume host mem and only host mem is always supported
2. seems this is an existing issue with the test, why did it not happen before this PR?

Reply:
1. That's UCP test which is working with many MDs. So do you mean to iterate somehow through all MDs and collect map from all the supported MDs?
2. That test had no `DEVX_OBJECTS` variable set, so ODP flag was disabled for IB MD.